src/mongocxx/collection.cpp

```diff
@@ -327,30 +327,24 @@ stdx::optional<result::delete_result> collection::delete_one(
     return stdx::optional<result::delete_result>(result::delete_result(std::move(result.value())));
 }
 
-stdx::optional<bsoncxx::document::value> collection::find_one_and_replace(
-    bsoncxx::document::view filter, bsoncxx::document::view replacement,
-    const options::find_one_and_replace& options) {
+stdx::optional<bsoncxx::document::value> _find_and_modify(::mongoc_collection_t* collection,
```

Comment:
If this is a free file-scoped function, it should go in an unnamed namespace and probably at the top of the file. You should also drop the _ from the beginning. Global names beginning with _[a-z] are reserved. You aren't quite global here because you are in the mongocxx namespace, but you are too close for comfort.